### ISSUE TITLE
[BUG] Un-closed transport in test suite.

### ISSUE THREAD

**Issue by dwoz** [saltstack/salt#65759]
```
/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/transport/base.py:281: TransportWarning: Unclosed transport! <salt.transport.tcp.PublishClient object at 0xffff8ca36aa0>
               File "/tmp/testing/artifacts/salt/lib/python3.10/runpy.py", line 196, in _run_module_as_main
                 return _run_code(code, main_globals, None,
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/runpy.py", line 86, in _run_code
                 exec(code, run_globals)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/coverage/__main__.py", line 10, in <module>
                 sys.exit(main())
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/coverage/cmdline.py", line 973, in main
                 status = CoverageScript().command_line(argv)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/coverage/cmdline.py", line 684, in command_line
                 return self.do_run(options, args)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/coverage/cmdline.py", line 861, in do_run
                 runner.run()
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/coverage/execfile.py", line 211, in run
                 exec(code, main_mod.__dict__)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pytest/__main__.py", line 5, in <module>
                 raise SystemExit(pytest.console_main())
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/config/__init__.py", line 189, in console_main
                 code = main()
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/config/__init__.py", line 166, in main
                 ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__
                 return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
                 res = hook_impl.function(*args)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/main.py", line 316, in pytest_cmdline_main
                 return wrap_session(config, _main)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/main.py", line 269, in wrap_session
                 session.exitstatus = doit(config, session) or 0
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/main.py", line 323, in _main
                 config.hook.pytest_runtestloop(session=session)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__
                 return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
                 res = hook_impl.function(*args)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/main.py", line 348, in pytest_runtestloop
                 item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__
                 return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
                 res = hook_impl.function(*args)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/flaky/flaky_pytest_plugin.py", line 94, in pytest_runtest_protocol
                 self.runner.pytest_runtest_protocol(item, nextitem)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/runner.py", line 114, in pytest_runtest_protocol
                 runtestprotocol(item, nextitem=nextitem)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/runner.py", line 127, in runtestprotocol
                 rep = call_and_report(item, "setup", log)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/flaky/flaky_pytest_plugin.py", line 138, in call_and_report
                 call = call_runtest_hook(item, when, **kwds)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/runner.py", line 261, in call_runtest_hook
                 return CallInfo.from_call(
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/runner.py", line 341, in from_call
                 result: Optional[TResult] = func()
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/runner.py", line 262, in <lambda>
                 lambda: ihook(item=item, **kwds), when=when, reraise=reraise
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__
                 return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
                 res = hook_impl.function(*args)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/runner.py", line 157, in pytest_runtest_setup
                 item.session._setupstate.setup(item)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/runner.py", line 494, in setup
                 col.setup()
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/python.py", line 1802, in setup
                 self._request._fillfixtures()
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 551, in _fillfixtures
                 item.funcargs[argname] = self.getfixturevalue(argname)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 570, in getfixturevalue
                 fixturedef = self._get_active_fixturedef(argname)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 592, in _get_active_fixturedef
                 self._compute_fixture_value(fixturedef)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 678, in _compute_fixture_value
                 fixturedef.execute(request=subrequest)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 1052, in execute
                 fixturedef = request._get_active_fixturedef(argname)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 592, in _get_active_fixturedef
                 self._compute_fixture_value(fixturedef)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 678, in _compute_fixture_value
                 fixturedef.execute(request=subrequest)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 1052, in execute
                 fixturedef = request._get_active_fixturedef(argname)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 592, in _get_active_fixturedef
                 self._compute_fixture_value(fixturedef)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 678, in _compute_fixture_value
                 fixturedef.execute(request=subrequest)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 1076, in execute
                 result = ihook.pytest_fixture_setup(fixturedef=self, request=request)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_hooks.py", line 265, in __call__
                 return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_manager.py", line 80, in _hookexec
                 return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
                 res = hook_impl.function(*args)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 1130, in pytest_fixture_setup
                 result = call_fixture_func(fixturefunc, request, kwargs)
           
               File "/tmp/testing/.nox/ci-test-onedir/lib/python3.10/site-packages/_pytest/fixtures.py", line 909, in call_fixture_func
                 fixture_result = fixturefunc(**kwargs)
           
               File "/tmp/testing/tests/pytests/integration/netapi/rest_tornado/conftest.py", line 21, in app
                 return netapi.build_tornado_app(
           
               File "/tmp/testing/tests/support/netapi.py", line 124, in build_tornado_app
                 application.event_listener = saltnado.EventListener(
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/netapi/rest_tornado/saltnado.py", line 276, in __init__
                 self.event = salt.utils.event.get_event(
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/utils/event.py", line 127, in get_event
                 return MasterEvent(
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/utils/event.py", line 928, in __init__
                 super().__init__(
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/utils/event.py", line 265, in __init__
                 self.connect_pub()
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/utils/event.py", line 348, in connect_pub
                 self.subscriber = salt.transport.ipc_publish_client(
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/transport/base.py", line 210, in ipc_publish_client
                 return publish_client(opts, io_loop, **kwargs)
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/transport/base.py", line 152, in publish_client
                 return salt.transport.tcp.PublishClient(
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/transport/tcp.py", line 219, in __init__
                 super().__init__(opts, io_loop, **kwargs)
           
               File "/tmp/testing/artifacts/salt/lib/python3.10/site-packages/salt/transport/base.py", line 398, in __init__
                 super().__init__()
           
               warnings.warn(
             Enable tracemalloc to get traceback where the object was allocated.
```

**Comment by dwoz** [comment]
Fixed in `3006.7`

**Comment by ironman627** [comment]
Running into this same issue in 3008.2.

jobcompletion leaks a PublishClient on every reconnect
**Symptom**: a Python ResourceWarning-style message surfaces only in journalctl/stderr (never in the master's own log file, since warnings.warn() bypasses the logging framework):

/opt/saltstack/salt/lib/python3.14/site-packages/salt/transport/base.py:310: TransportWarning: Unclosed transport! <salt.transport.tcp.PublishClient object at 0x7f3c67100d60>
  warnings.warn(...)
TransportWarning: Enable tracemalloc to get the object allocation traceback
**Root cause**: traced via full traceback frames to sseape/engines/jobcompletion.py:210 → salt.utils.event.get_master_event() → MasterEvent.__init__ → connect_pub() → opens a new salt.transport.tcp.PublishClient (via salt.utils.asynchronous.SyncWrapper) to subscribe to the master's event bus. The previous PublishClient instance from an earlier subscribe is never explicitly closed before being replaced/garbage collected. Every time jobcompletion reconnects its event subscriber, one more PublishClient (and whatever OS resources it holds) leaks.

**Evidence**: full traceback (from journalctl -u salt-master, timestamped the same second as an unplanned master restart with no preceding clean-shutdown/SIGTERM sequence in the master's own log) showing the exact call chain above, ending in the TransportWarning.

```
[root@salt-raas ~]# salt \* test.version 
web-app-salt:
    3008.2
saltmaster:
    3008.2
```
